restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisIncomingPhoneNumbersDao.java

```diff
@@ -70,19 +74,61 @@ public IncomingPhoneNumber getIncomingPhoneNumber(final String phoneNumber) {
         return getIncomingPhoneNumber("getIncomingPhoneNumberByValue", phoneNumber);
     }
 
-    private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
+   private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
         final SqlSession session = sessions.openSession();
-        try {
+        String inboundPhoneNumber = null;
+          try {
+            //if(parameter.toString().contains("*")){}{
+            //perform direct check
             final Map<String, Object> result = session.selectOne(namespace + selector, parameter);
-            if (result != null) {
+            if (result != null ) {
                 return toIncomingPhoneNumber(result);
-            } else {
-                return null;
-            }
-        } finally {
+            }//}
+            //check if there is a Regex match only if parameter is a String aka phone Number
+            if(!(parameter instanceof Sid)){
+               inboundPhoneNumber = parameter.toString().replace("+1", "");
+               logger.warn(" Inbound phoneNumber : " + inboundPhoneNumber );
+              return checkIncomingPhoneNumberRegexMatch(selector, inboundPhoneNumber);
+               }
+
+        }finally {
+                     session.close();
+         }
+          return null;
+
+}
+
+   public IncomingPhoneNumber checkIncomingPhoneNumberRegexMatch ( String selector, String inBoundPhoneNumber){
+               final SqlSession session = sessions.openSession();
+               String phoneRegexPattern = null;
+               try {
+                    List<IncomingPhoneNumber> listPhones = getAllIncomingPhoneNumbers();
+                   for (IncomingPhoneNumber listPhone : listPhones){
```

Comment:
In order to avoid iterating over ALL IncomingPhoneNumbers we should add a new column at the IncomingPhoneNumbers table to flag REGEX true/false when we create a DID.
There must be a validation method to tell if the number is DID, USSD ShortCode or Regex